worker-message/src/worker.rs

```diff
@@ -0,0 +1,325 @@
+use crate::message::{ExecuteOutput, Request, Response};
+use crate::{
+    CoordinatorMessage, ExecuteCommandRequest, ExecuteCommandResponse, JobReport, ReadFileRequest,
+    ReadFileResponse, ResponseResult, StreamCommandRequest, StreamCommandResponse, WorkerMessage,
+    WriteFileRequest, WriteFileResponse,
+};
+use std::path::{Path, PathBuf};
+use std::process::{Output, Stdio};
+
+use anyhow::{Error, Result};
+use std::time::SystemTime;
+use tempdir::TempDir;
+use tokio::fs;
+use tokio::io::{AsyncBufReadExt, AsyncReadExt, AsyncWriteExt, BufReader};
+use tokio::process::{Child, Command};
+use tokio::sync::mpsc;
+use tokio::task::{JoinHandle, JoinSet};
+use tracing::info;
+
+pub async fn listen() {
+    let mut tasks = JoinSet::new();
+    let (tx, mut coordinator_rx) = mpsc::channel(8);
+    tasks.spawn(async move {
+        let mut buffer = Vec::new();
+        let mut input = tokio::io::stdin();
+        loop {
+            let payload_len = input
+                .read_u64()
+                .await
+                .expect("Failed to read standard input");
```

Comment:
But when a `worker` cannot read from its stdin, the only choice left is to exit with a message.
Does it still make sense to define a new `Error` variant in this case?

Reply:
See my other comments about how I don’t replace every `unwrap`, but I still err on the side of errors. That way I can bubble the error back up to main and then do something consistent. For example, here we might log the error via tracing instead of raw stderr so that it’s consistent with other log messages.

Reply:
Good example. I got the benefit.